hs_core/hydroshare/resource.py

```diff
@@ -1030,6 +1095,30 @@ def resolve_minid(minid):
     return utils.get_resource_by_minid(minid).short_id
 
 
+def resolve_doi(doi):
+    """
+    Takes as input a DOI and returns the internal CommonsShare identifier (pid) for a resource.
+    This method will be used to get the CommonsShare pid for a resource identified by a doi for
+    further operations using the web service API.
+
+    REST URL:  GET /resolveDOI/{doi}
+
+    Parameters:    doi - A doi assigned to a resource in CommonsShare.
+
+    Returns:    The doi of the resource that was published
+
+    Return Type:    pid
+
+    Raises:
+    Exceptions.NotAuthorized - The user is not authorized
+    Exceptions.NotFound - The resource identified by pid does not exist
+    Exception.ServiceFailure - The service is unable to process the request
+
+    Note:  All CommonsShare methods (except this one) will use CommonsShare internal identifiers
+    (pids). This method exists so that a program can resolve the pid for a DOI.
+    """
+    return utils.get_resource_by_doi(doi).short_id
```

Comment:
What happens if there is no corresponding resource for an input doi?   It'd be more robust to break this single return statement up into a conditional statement to only return short_id when a valid resource is retrieved from DOI, otherwise, return None or something.

Reply:
After looking at get_resource_by_doi() method, I think it may work better to combine get_resource_by_doi() and resolve_doi() into one function, but since get_resource_by_doi() will return 404 exception, so I think it covered my question above regarding the scenario where there is no resource for the input doi